src/lib/support/jsontlv/README.md

```diff
@@ -0,0 +1,106 @@
+
+In order for the Json format to represent the TLV format without loss of
+information, the Json key of each element should contain the TLV element tag and
+type information.
+
+The Json key schema is as follows:
+
+```
+{ [field_name:]field_id:element_type[-sub_element_type] }
+```
+
+Specific rules:
+
+-   'element_type' MUST map to the TLV Type as specified in the table below.
+-   For 'element_type' of type "ARRAY", the sub_element_type MUST always be
+    present and be equal to the 'element_type' of the elements in the array.
+-   Array elements MUST have the same 'element_type'.
+-   Array elements MUST have anonymous TLV tags.
+-   'field_name' is always an optional field and can be provided if the semantic
+    name for the field is available.
+-   In the case of an empty array, 'sub_element_type' should be "?" (unknown
+    element type).
+-   The unknown element type “?” MUST never occur outside of "ARRAY-?".
+-   'field_id' is equivalent to the Field ID for all cluster payloads (commands,
+    events, attributes).
+
+The table below summarizes all element types and their corresponding encoding in
+the Json key:
+
+| TLV Type                               | element_type           | JSON Type               |
+| -------------------------------------- | ---------------------- | ----------------------- |
+| Unsigned Integer ( < 2^32 )            | UINT                   | Number                  |
+| Unsigned Integer ( >= 2^32 )           | UINT                   | String                  |
+| Signed Integer ( >= -2^31 and < 2^31 ) | INT                    | Number                  |
+| Signed Integer ( < -2^31 or >= 2^31 )  | INT                    | String                  |
+| Boolean                                | BOOL                   | Boolean                 |
+| Float (32-bit)                         | FLOAT                  | Number                  |
+| Float/Double (64-bit)                  | DOUBLE                 | Number                  |
+| Octet String                           | BYTES                  | String (Base64 encoded) |
+| UTF-8 String                           | STRING                 | String                  |
+| Null Value                             | NULL                   | null                    |
+| Struct                                 | STRUCT                 | Dict                    |
+| Array                                  | ARRAY-sub_element_type | Array                   |
+| Unknown                                | ?                      | Empty Array             |
+
+Note that this is NOT a generalized JSON representation of TLV and does not
+support arbitrary TLV conversions. The main goal of this Json format is to
+support data model payloads for events/commands/attributes. Some of the
+limitations of this format are:
+
+-   TLV List types are not supported.
+-   TLV Array cannot contain another TLV Array.
```

Comment:
That will be an issue to convert some of the consumers. The current code does support array nesting and beside pure logging the certification tests are relying on such behaviour.